core/iwasm/compilation/aot_emit_numberic.c

```diff
@@ -171,6 +171,15 @@
         right = shift_count_mask;                                      \
     } while (0)
 
+static bool
+is_shift_count_mask_needed(AOTCompContext *comp_ctx, LLVMValueRef left,
+                           LLVMValueRef right)
+{
+    return (strcmp(comp_ctx->target_arch, "x86_64") != 0
+            && strcmp(comp_ctx->target_arch, "i386") != 0)
+           || (LLVMIsAConstant(left) && LLVMIsAConstant(right));
```

Comment:
my suggestion is using `LLVMIsEfficientConstInt()` here. The new function is able to make sure both `left` and `right` are *constant int* only.